packages/core-mobile/app/services/fcm/FCMService.ts

```diff
@@ -0,0 +1,82 @@
+import messaging, {
+  FirebaseMessagingTypes
+} from '@react-native-firebase/messaging'
+import Logger from 'utils/Logger'
+import NotificationsService from 'services/notifications/NotificationsService'
+import { ChannelId } from 'services/notifications/channels'
+import { ACTIONS, PROTOCOLS } from 'contexts/DeeplinkContext/types'
+
+type UnsubscribeFunc = () => void
+
+/**
+ * Wrapper for @react-native-firebase/messaging
+ */
+class FCMService {
+  getFCMToken = async (): Promise<string> => {
+    return await messaging().getToken()
+  }
+
+  listenForMessagesForeground = (): UnsubscribeFunc => {
+    return messaging().onMessage(async remoteMessage => {
+      Logger.info('A new FCM message arrived!', remoteMessage)
+      if (this.isValidRemoteMessage(remoteMessage)) {
+        //TO DESIGN: show in-app notification instead of native
+        const data = {
+          accountAddress: remoteMessage.data?.accountAddress ?? '',
+          chainId: remoteMessage.data?.chainId ?? '',
+          transactionHash: remoteMessage.data?.transactionHash ?? '',
+          url: `${PROTOCOLS.CORE}://${ACTIONS.OpenChainPortfolio}`
+        }
```

Comment:
let's use zod schema to validate the remote message. you will then not need to do `?`

Reply:
https://github.com/ava-labs/core-mobile/pull/1659/commits/f6f59c15f6176506201760f557e5a15288dff150